# Enforce having expectation in test body (`vitest/expect-expect`)

ðŸ’¼ This rule is enabled in the âœ… `recommended` config.

<!-- end auto-generated rule header -->

## Rule Details

Examples of **incorrect** code for this rule:

```js
test('myLogic', () => {
	const actual = myLogic()
})
```

Examples of **correct** code for this rule:

```js
test('myLogic', () => {
  const actual = myLogic()
  expect(actual).toBe(true)
})
```

## Options

> Default: `expect`

Array of custom expression strings that are converted into a regular expression.

```json
{
  "custom-expressions": [
    "expectValue",
    "mySecondExpression"
  ]
}
```
