# Disallow conditional tests (`@vitest/no-conditional-tests`)

âš ï¸ This rule _warns_ in the ğŸŒ `all` config.

<!-- end auto-generated rule header -->

## Rule Details

Examples of **incorrect** code for this rule:

```js
describe('my tests', () => {
	if (true) {
		it('is awesome', () => {
			doTheThing()
		})
	}
})
```

Examples of **correct** code for this rule:

```js
describe('my tests', () => {
	if (Math.random() > 0.5) {
		it('is awesome', () => {
			doTheThing()
		})
	}
})
```
