# vitest/no-disabled-tests

ğŸ“ Disallow disabled tests.

âš ï¸ This rule _warns_ in the following configs: ğŸŒ `all`, âœ… `recommended`.

<!-- end auto-generated rule header -->

## Rule Details

This rule disallows disabled tests.

Examples of **incorrect** code for this rule:

```ts
test.skip('foo', () => {
  // ...
})
```

Examples of **correct** code for this rule:

```ts
test('foo', () => {
  // ...
})
```
