# no-identical-title

ğŸ’¼âš ï¸ This rule is enabled in the â˜‘ï¸ `legacy-recommended` config. This rule _warns_ in the ğŸ’¾ `legacy-all` config.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## Rule Details

Examples of **incorrect** code for this rule:

```js
it('is awesome', () => {
	/* ... */
})

it('is awesome', () => {
	/* ... */
})
```

Examples of **correct** code for this rule:

```js
it('is awesome', () => {
	/* ... */
})

it('is very awesome', () => {
	/* ... */
})
```
