# Disallow skipped tests (`vitest/no-skipped-tests`)

âš ï¸ This rule _warns_ in the ğŸŒ `all` config.

<!-- end auto-generated rule header -->

## Rule Details

Examples of **incorrect** code for this rule:

```js
it.skip('test', () => {
	/* ... */
})

test.skip('it', () => {
	/* ... */
})
```

Examples of **correct** code for this rule:

```js
it('test', () => {
	/* ... */
})

test('it', () => {
	/* ... */
})
```
