# Disallow return statements in tests (`@vitest/no-test-return-statement`)

âš ï¸ This rule _warns_ in the ğŸŒ `all` config.

<!-- end auto-generated rule header -->

### Rule Details

incorrect code for this rule:

```ts
import { test } from 'vitest'

test('foo', () => {
  return expect(1).toBe(1)
})
```

correct code for this rule:

```ts
import { test } from 'vitest'

test('foo', () => {
  expect(1).toBe(1)
})
```
