# vitest/prefer-hooks-on-top

ğŸ“ Enforce having hooks before any test cases.

âš ï¸ This rule _warns_ in the ğŸŒ `all` config.

<!-- end auto-generated rule header -->

```ts
// bad

describe('foo', () => {
  it('bar', () => {
    // ...
  })

  beforeEach(() => {
    // ...
  })
})

// good

describe('foo', () => {
  beforeEach(() => {
    // ...
  })

  it('bar', () => {
    // ...
  })
})
```
