# vitest/prefer-to-be-truthy

ğŸ“ Enforce using `toBeTruthy`.

ğŸš« This rule is _disabled_ in the ğŸŒ `all` config.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

```js
// bad
expect(foo).toBe(true)
expectTypeOf(foo).toBe(true)

// good
expect(foo).toBeTruthy()
expectTypeOf(foo).toBeTruthy()
```
