# Suggest using `toBeTruthy` (`vitest/prefer-to-be-truthy`)

<<<<<<< HEAD
âš ï¸ This rule _warns_ in the ðŸŒ `all` config.

=======
>>>>>>> d24de5b (fix plugin key (#409))
ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

```js
// bad
expect(foo).toBe(true)
expectTypeOf(foo).toBe(true)

// good
expect(foo).toBeTruthy()
expectTypeOf(foo).toBeTruthy()
```
