# vitest/prefer-hooks-in-order

ğŸ“ Enforce having hooks in consistent order.

âš ï¸ This rule _warns_ in the ğŸŒ `all` config.

<!-- end auto-generated rule header -->

```js
// consistent order of hooks
;['beforeAll', 'beforeEach', 'afterEach', 'afterAll']
```

```js
// bad
afterAll(() => {
  removeMyDatabase()
})
beforeAll(() => {
  createMyDatabase()
})
```

```js
// good
beforeAll(() => {
  createMyDatabase()
})
afterAll(() => {
  removeMyDatabase()
})
```
